src/tests/unit/platform/test_users.py

```diff
@@ -0,0 +1,217 @@
+    assert user.email == 'jsheridan@zhadum.net'
+
+
+def test_get_and_delete_user(cbcsdk_mock):
+    """Tests retrieving and deleting a user."""
+    cbcsdk_mock.mock_request('GET', '/appservices/v6/orgs/test/users', GET_USERS_RESP)
+    cbcsdk_mock.mock_request('DELETE', '/appservices/v6/orgs/test/users/3978', None)
+    api = cbcsdk_mock.api
+    user = api.select(User, 3978)
+    assert user.last_name == 'Mariner'
+    assert user.first_name == 'Beckett'
+    assert user.login_name == 'bmariner@cerritos.starfleet.mil'
+    assert user.email == 'bmariner@cerritos.starfleet.mil'
+    assert user.urn == 'psc:user:test:3978'
+    user.delete()
+
+
+def test_get_user_and_reset_googleauth(cbcsdk_mock):
+    """Tests retrieving a user and resetting Google authentication."""
+    cbcsdk_mock.mock_request('GET', '/appservices/v6/orgs/test/users', GET_USERS_RESP)
+    cbcsdk_mock.mock_request('DELETE', '/appservices/v6/orgs/test/users/4338/googleauth', None)
+    api = cbcsdk_mock.api
+    user = api.select(User, 4338)
+    assert user.last_name == 'Jackson'
+    assert user.first_name == 'Daniel'
+    assert user.login_name == 'djackson@stargate-command.af.mil'
+    assert user.email == 'djackson@stargate-command.af.mil'
+    assert user.urn == 'psc:user:test:4338'
+    user.reset_google_authenticator_registration()
+
+
+def test_get_nonexistent_user(cbcsdk_mock):
+    """Test retrieving a nonexistent user."""
+    cbcsdk_mock.mock_request('GET', '/appservices/v6/orgs/test/users', GET_USERS_RESP)
+    api = cbcsdk_mock.api
+    with pytest.raises(ObjectNotFoundError):
+        api.select(User, 999)
+
+
+def test_attempt_to_create_naively(cb):
+    """Attempt to create a user by just creating one and saving it, which doesn't work."""
+    user = User(cb, None)
+    user.last_name = 'Rios'
+    user.first_name = 'Cristobal'
+    user.login_name = 'rios@la-sirena.net'
+    user.email = 'rios@la-sirena.net'
+    with pytest.raises(ApiError):
+        user.save()
+
+
+def test_create_user(cbcsdk_mock):
+    """Test creating a user."""
+    post_made = False
+
+    def check_post(uri, body, **kwargs):
+        assert body == EXPECT_USER_ADD
+        nonlocal post_made
+        post_made = True
+        return USER_ADD_SUCCESS_RESP
+
+    def check_get(uri, query_params, default):
+        nonlocal post_made
+        assert post_made
+        return GET_USERS_AFTER_CREATE_RESP
+
+    cbcsdk_mock.mock_request('POST', '/appservices/v6/orgs/test/users', check_post)
+    cbcsdk_mock.mock_request('GET', '/appservices/v6/orgs/test/users', check_get)
+    api = cbcsdk_mock.api
+    builder = User.create(api).set_email('rios@la-sirena.net').set_role('psc:role:test:APP_SERVICE_ROLE')
+    builder.set_first_name('Cristobal').set_last_name('Rios')
+    builder.add_grant_profile(['psc:org:test2'], ['psc:role:test2:DUMMY'])
+    builder.add_grant_profile(['test3'], ['psc:role:test3:DUMMY'])
+    user, password = builder.build()
```

Comment:
Have we confirmed this behavior as my understanding is that the creation of the user requires the user to receive the email and fill out their own password. I am not aware of a temporary password that is useable

Reply:
Correct.

Reply:
Taking out return of the password, as per @kebringer-cb 's discussion.